queries/nu/highlights.scm

```diff
@@ -0,0 +1,396 @@
+  "const"
+  "hide-env"
+  "source"
+  "source-env"
+  "overlay"
+  "register"
+  "loop"
+  "while"
+  "error"
+  "do"
+  "if"
+  "else"
+  "try"
+  "catch"
+  "match"
+  "break"
+  "continue"
+  "return"
+] @keyword
+
+(hide_mod
+  "hide" @keyword)
+
+(decl_use
+  "use" @keyword)
+
+(ctrl_for
+  "for" @keyword
+  "in" @keyword)
+
+(overlay_list
+  "list" @keyword.import)
+
+(overlay_hide
+  "hide" @keyword.import)
+
+(overlay_new
+  "new" @keyword.import)
+
+(overlay_use
+  "use" @keyword.import
+  "as" @keyword)
+
+(ctrl_error
+  "make" @keyword.import)
+
+; ---
+; literals
+(val_number) @number
+
+(val_duration
+  unit: _ @variable.parameter)
+
+(val_filesize
+  unit: _ @variable.parameter)
+
+(val_binary
+  [
+    "0b"
+    "0o"
+    "0x"
+  ] @number
+  "[" @punctuation.bracket
+  digit: [
+    "," @punctuation.delimiter
+    (hex_digit) @number
+  ]
+  "]" @punctuation.bracket) @number
+
+(val_bool) @constant.builtin
+
+(val_nothing) @constant.builtin
+
+(val_string) @string
+
+arg_str: (val_string) @variable.parameter
+
+file_path: (val_string) @variable.parameter
+
+(val_date) @number
+
+(inter_escape_sequence) @string.escape
+
+(escape_sequence) @string.escape
+
+(val_interpolated
+  [
+    "$\""
+    "$\'"
+    "\""
+    "\'"
+  ] @string)
+
+(unescaped_interpolated_content) @string
+
+(escaped_interpolated_content) @string
+
+(expr_interpolated
+  [
+    "("
+    ")"
+  ] @variable.parameter)
+
+; ---
+; operators
+(expr_binary
+  [
+    "+"
+    "-"
+    "*"
+    "/"
+    "mod"
+    "//"
+    "++"
+    "**"
+    "=="
+    "!="
+    "<"
+    "<="
+    ">"
+    ">="
+    "=~"
+    "!~"
+    "and"
+    "or"
+    "xor"
+    "bit-or"
+    "bit-xor"
+    "bit-and"
+    "bit-shl"
+    "bit-shr"
+    "in"
+    "not-in"
+    "starts-with"
+    "ends-with"
+  ] @operator)
+
+(where_command
+  [
+    "+"
+    "-"
+    "*"
+    "/"
+    "mod"
+    "//"
+    "++"
+    "**"
+    "=="
+    "!="
+    "<"
+    "<="
+    ">"
+    ">="
+    "=~"
+    "!~"
+    "and"
+    "or"
+    "xor"
+    "bit-or"
+    "bit-xor"
+    "bit-and"
+    "bit-shl"
+    "bit-shr"
+    "in"
+    "not-in"
+    "starts-with"
+    "ends-with"
+  ] @operator)
+
+(assignment
+  [
+    "="
+    "+="
+    "-="
+    "*="
+    "/="
+    "++="
+  ] @operator)
+
+(expr_unary
+  [
+    "not"
+    "-"
+  ] @operator)
+
+(val_range
+  [
+    ".."
+    "..="
+    "..<"
+  ] @operator)
+
+[
+  "=>"
+  "="
+  "|"
+] @operator
+
+[
+  "o>"
+  "out>"
+  "e>"
+  "err>"
+  "e+o>"
+  "err+out>"
+  "o+e>"
+  "out+err>"
+] @operator
+
+; ---
+; punctuation
+[
+  ","
+  ";"
+] @punctuation.special
+
+(param_long_flag
+  "--" @punctuation.delimiter)
+
+(long_flag
+  "--" @punctuation.delimiter)
+
+(long_flag_equals_value
+  "--" @punctuation.delimiter)
+
+(short_flag
+  "-" @punctuation.delimiter)
+
+(long_flag_equals_value
+  "=" @punctuation.special)
+
+(param_short_flag
+  "-" @punctuation.delimiter)
+
+(param_rest
+  "..." @punctuation.delimiter)
+
+(param_type
+  ":" @punctuation.special)
```

Comment:
I think all of these special highlights should be replaced with a top level `":" @punctuation.delimiter` highlight, personally